lib/rspec/core/formatters.rb

```diff
@@ -170,6 +170,11 @@ def add(formatter_to_use, *paths)
       end
     end
 
+    # @private
+    def change_output_stream(output_stream)
+      @formatters.each { |formatter| formatter.change_output(output_stream) }
```

Comment:
This needs to cope with formatters not implementing this method until RSpec 4, as it's a breaking change in our API. Formatters can be defined in many ways not just inheriting from our classes.